xline/src/storage/snapshot_allocator.rs

```diff
@@ -0,0 +1,28 @@
+use std::error::Error;
+
+use async_trait::async_trait;
+use curp::SnapshotAllocator;
+use engine::snapshot_api::{MemorySnapshot, RocksSnapshot, SnapshotProxy};
+
+/// Rocks snapshot allocator
+pub(crate) struct RocksSnapshotAllocator;
+
+#[async_trait]
+impl SnapshotAllocator for RocksSnapshotAllocator {
+    async fn allocate_new_snapshot(&self) -> Result<SnapshotProxy, Box<dyn Error>> {
+        let tmp_path = format!("/tmp/snapshot-{}", uuid::Uuid::new_v4());
```

Comment:
What's stored in this `tmp_path`f? Snapshot files? Should this `path` be configurable?

Reply:
This interface is an empty snapshot created by the follower to receive the data sent by the leader, after follower apply the snapshot, these files are no longer useful, and will be deleted